### PR TITLE
[DOCS] Privileges Matrix for Detections Feature 

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Addresses #536 [here](https://security-docs_634.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html).

[Preview link](https://security-docs_634.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html).

### CODE REVIEW

File: docs/getting-started/detections-req.asciidoc [hunk 1]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
 IMPORTANT: After changing the `xpack.encryptedSavedObjects.encryptionKey` value
 and restarting {kib}, you must restart all detection rules.
 
+[discrete]
+[[enable-and-access-detections]]
+== Enable and Access Detections
+Enabling the Detections feature allows you to view and use the Detections page from within the Security app. You only need to complete this action once and can choose between enabling it for a single Kibana space or across multiple ones. Once the Detections feature is enabled, additional privileges are needed to use Detection rules and alerts. The table below explains how to enable the Detections feature and describes required privileges for viewing and using rules and alerts.
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:41]:
```suggestion
Enabling the Detections feature allows you to view and use the Detections page within the Security app. You only need to complete this action once and can choose between enabling it for a single Kibana space or across multiple ones. Once the Detections feature is enabled, additional privileges are needed to manage detection rules and alerts, described in the following table. 
```

File: docs/getting-started/detections-req.asciidoc [hunk 2]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+a| * The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the for the following system indices where <kib-space>` is the {kib} space name:
+         ** `.siem-signals-<kib-space>`
+         ** `.lists-<kib-space>`
+         ** `.items-<kib-space>`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+|Manage alerts
+
+Note: Allows you to manage alerts, but not modify rules.
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:81]:
```suggestion
**Note:** Allows you to manage alerts, but not modify rules.
```

File: docs/getting-started/detections-req.asciidoc [hunk 3]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+
+|Enable the Detections feature in all Kibana spaces
+|The `manage` privilege
+a| * The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices:
+         ** `.siem-signals-*`
+         ** `.lists-*`
+         ** `.items-*` system indices.
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:66]:
```suggestion
         ** `.items-*`
```

File: docs/getting-started/detections-req.asciidoc [hunk 4]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+
+[discrete]
+[width="100%",options="header"]
+|==============================================
+|Action |Cluster Privileges |Index Privileges |Kibana Privileges
+
+|Enable the Detections feature in your Kibana space
+|The `manage` privilege
+
+a| * The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:53]: I think we can remove the first bullet. 

```suggestion
a| The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```

File: docs/getting-started/detections-req.asciidoc [hunk 5]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+         ** `.siem-signals-<kib-space>`
+         ** `.lists-<kib-space>`
+         ** `.items-<kib-space>`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+
+|Enable the Detections feature in all Kibana spaces
+|The `manage` privilege
+a| * The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:63]:
```suggestion
a| The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices:
```

File: docs/getting-started/detections-req.asciidoc [hunk 6]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+         ** `.siem-signals-*`
+         ** `.lists-*`
+         ** `.items-*` system indices.
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+|Manage rules
+| N/A
+a| * The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:72]:
```suggestion
a| The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the for the following system indices where <kib-space>` is the {kib} space name:
```

File: docs/getting-started/detections-req.asciidoc [hunk 7]
```diff
@@ -35,6 +35,61 @@ of at least 32 characters. For example:
...
+         ** `.lists-<kib-space>`
+         ** `.items-<kib-space>`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+|Manage alerts
+
+Note: Allows you to manage alerts, but not modify rules.
+|N/A
+a| * The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:83]:
```suggestion
a| The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```

**Review by jmikell821 — CHANGES_REQUESTED** [review]
Hi @nastasha-solomon - few slight suggestions. Please also fix this grammatical error in the beginning. 
image

**Comment by nastasha-solomon** [comment]
@jmikell821 updates are ready for your review when you have a chance!

File: docs/getting-started/detections-req.asciidoc [hunk 8]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
 IMPORTANT: After changing the `xpack.encryptedSavedObjects.encryptionKey` value
 and restarting {kib}, you must restart all detection rules.
 
+[discrete]
+[[enable-and-access-detections]]
+== Enable and Access Detections
+Enabling the Detections feature allows you to view and use the Detections page from within the Security app. You only need to complete this action once and can choose between enabling it for a single Kibana space or across multiple ones. Once the Detections feature is enabled, additional privileges are needed to use Detection rules and alerts, described in the following table.
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:41]:
```suggestion
Enabling the Detections feature allows you to view and use the Detections page from within the Security app. You only need to complete this action once and can choose between enabling it for a single Kibana space or across multiple ones. Once the Detections feature is enabled, additional privileges are needed to use detection rules and alerts, described in the following table.
```

File: docs/getting-started/detections-req.asciidoc [hunk 9]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
...
+
+[discrete]
+[width="100%",options="header"]
+|==============================================
+|Action |Cluster Privileges |Index Privileges |Kibana Privileges
+
+|Enable the Detections feature in your Kibana space
+|The `manage` privilege
+
+a|The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:53]: Let's make `<kib-space>` code text throughout. 

```suggestion
a|The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where `<kib-space>` is the {kib} space name:
```

File: docs/getting-started/detections-req.asciidoc [hunk 10]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
...
+
+* `.siem-signals-*`
+* `.lists-*`
+* `.items-*`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+|Manage rules
+| N/A
+a|The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:74]:
```suggestion
a|The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the for the following system indices where `<kib-space>` is the {kib} space name:
```

File: docs/getting-started/detections-req.asciidoc [hunk 11]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
...
+* `.lists-<kib-space>`
+* `.items-<kib-space>`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
+
+|Manage alerts
+
+**Note**: Allows you to manage alerts, but not modify rules.
+|N/A
+a|The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where <kib-space>` is the {kib} space name:
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:86]:
```suggestion
a|The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices where `<kib-space>` is the {kib} space name:
```

**Comment by jmikell821** [comment]
Hi @spong and @peluja1012 - please review this draft matrix privileges PR. We left the original paragraphs in the topic for now as a reference point, but will delete them before we publish. We'll be enhancing this page as Security enables feature-based permissions, but for now, please take a look and let us know if there's anything missing or incorrect. Thank you!

**Review by jmikell821 — APPROVED** [review]

File: docs/getting-started/detections-req.asciidoc [hunk 12]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
 IMPORTANT: After changing the `xpack.encryptedSavedObjects.encryptionKey` value
 and restarting {kib}, you must restart all detection rules.
 
+[discrete]
+[[enable-and-access-detections]]
+== Enable and Access Detections
+Enabling the Detections feature allows you to view and use the Detections page from within the Security app. You only need to complete this action once and can choose between enabling it for a single Kibana space or across multiple ones. Once the detections feature is enabled, additional privileges are needed to use Detection rules and alerts, described in the following table.
```
Comment (spong) [inline docs/getting-started/detections-req.asciidoc:41]: > You only need to complete this action once and can _choose between enabling it for a single Kibana space or across multiple ones._

Not sure I'm following the distinction here?  This action will need to be completed on each space they intend to enable detections, as there is no way to bulk setup multiple spaces via the UI (can be automated via the API though). Does 'choose between enabling...' mean here to "use this role and go to each space to complete setup"?

File: docs/getting-started/detections-req.asciidoc [hunk 13]
```diff
@@ -35,6 +35,65 @@ of at least 32 characters. For example:
...
+
+|Enable the Detections feature in all Kibana spaces
+|The `manage` privilege
+a|The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following system indices:
+
+* `.siem-signals-*`
+* `.lists-*`
+* `.items-*`
+|{kib} space `All` privileges for the `Security` feature (see
+{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
```
Comment (spong) [inline docs/getting-started/detections-req.asciidoc:70]: Ahhh, I think I see the distinction from above. Breaking out the `per space` vs `all space` access to alerts. This makes sense to me, but just note that even with the above `all space` configuration the user will need to go to the security app of each space to complete the `setup` of the detection engine (creating the index/ILM, etc).

**Review by spong — APPROVED** [review]
Just one nit about the distinction between `access` and `setup` between all spaces, but other than that LGTM! Thank you so much for tackling this one and making a clean and clear matrix to determine what privileges are necessary for specific actions! This is great for our users! 🙂 🎉

**Comment by spong** [comment]
Just noticed this @nastasha-solomon, but as of https://github.com/elastic/kibana/issues/90715 we don't need `Saved Objects Management`  `all` privileges anymore, so we can remove those two references from the Enable/Access Detections sections.

**Comment by jmikell821** [comment]
@nastasha-solomon here's my take on the intro para you had. Let's merge this in and see what it looks like. I'll comment out the superfluous text at the bottom afterwards. 

> To use the Detections page, the Detections feature must be enabled and your role must have access to rules and alerts. If your role does not have the cluster and index privileges needed to enable this feature, you can request someone who has these privileges to turn it on for you. The following table describes the required privileges to access the Detections page - including rules and alerts.

**Review by nastasha-solomon — COMMENTED** [review]
LGTM!

**Comment by jmikell821** [comment]
Hi @spong, @peluja1012 and @FrankHassanabad - would appreciate your review/feedback of this topic that's been heavily revised and reorganized. Please confirm everything is accurate and if you have any further suggestions. 

We'll be updating this topic regularly to coordinate with the new RBAC permissions. Thank you!

File: docs/getting-started/detections-req.asciidoc [hunk 14]
```diff
@@ -37,6 +37,70 @@ and restarting {kib}, you must restart all detection rules.
 
 [discrete]
 [[enable-detections-ui]]
+== Enable and access detections
+To use the Detections page, the Detections feature must be enabled and your role must have access to rules and alerts. If your role does not have the cluster and index privileges needed to enable this feature, you can request someone who has these privileges to turn it on for you. The following table describes the required privileges to access the Detections page - including rules and alerts.
```
Comment (spong) [inline docs/getting-started/detections-req.asciidoc:41]: Similar to my previous comment below, I think it's worth explicitly calling out that it's not just these privileges that are necessary to enable the Detections feature, but also that the user with the specified privileges _must visit the Security Application at least once_ for each space they'd like to enable the feature. We've had some [discuss posts](https://discuss.elastic.co/t/cant-access-detections-from-a-different-space/272118/9) where this aspect of the configuration wasn't entirely clear, so would be worth highlighting in a tip/info callout.

**Review by spong — APPROVED** [review]
Latest changes LGTM! Thanks for all the effort here @nastasha-solomon and @jmikell821! 🙂

**Pull request merged**